### PR TITLE
[docs] Fix some charts horizontal overflow on mobile devices

### PR DESCRIPTION
**Pull request by cupok**

### CODE REVIEW

**Review by alexfauquette — APPROVED** [review]
Thank you very much for the improvement of the documentation on mobile that's a nice improvement 🎉

To support both mobile and desktop, I think using the responsive width of the charts is more appropriate. Let me now if you have some opposition ot the modification I did

File: packages/x-charts/src/ResponsiveChartContainer/index.tsx [hunk 1]
```diff
@@ -96,6 +96,7 @@ const ResizableContainer = styled('div', {
   name: 'MuiResponsiveChart',
   slot: 'Container',
 })<{ ownerState: Pick<ResponsiveChartContainerProps, 'width' | 'height'> }>(({ ownerState }) => ({
+  maxWidth: '100%',
```
[the remark this answers is not in the document]
Comment (cupok) [reply packages/x-charts/src/ResponsiveChartContainer/index.tsx]: Yup! You're right, nice solution!

**Comment by cupok** [comment]
`To support both mobile and desktop, I think using the responsive width of the charts is more appropriate. Let me now if you have some opposition ot the modification I did`

I think in cases like "StackOrderDemo", where chart doesn't quite fit, fixed-width with "overflow: hidden;" may be appropriate

**Comment by alexfauquette** [comment]
Thanks, I've updated it a  bit to have the scroll only on the charts. I will merge it when tests pass 🚀

**Comment by cupok** [comment]
All looks pretty good now, good job!

**Pull request merged**